ldap/servers/slapd/back-ldbm/filterindex.c

```diff
@@ -899,11 +903,12 @@ list_candidates(
              * If we encounter a zero length idl, we bail now because this can never
              * result in a meaningful result besides zero.
              */
+            sr->sr_flags |= SR_FLAG_MUST_APPLY_FILTER_TEST;
```

Comment:
Because in the future we may add other optimisations to the OR case, that would still require filter eval when there are partial sets. Also remember, that provided the candidate set is less than test threshold, we early return and thus MUST do a filter test.

Reply:
Which by the way is the current behaviour, it's just that now we flag this in the filterindex.c code instead, because before it was flagged by lookup_returned_allidsp in onelevel_candidates().